src/main/java/intellistart/interviewplanning/model/period/TimeService.kt

```diff
@@ -0,0 +1,51 @@
+package intellistart.interviewplanning.model.period
+
+import intellistart.interviewplanning.model.user.UserService
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+import org.springframework.stereotype.Component
+import java.time.Duration
+import java.time.LocalTime
+import java.time.format.DateTimeFormatter
+import java.time.format.DateTimeParseException
+
+/**
+ * Utils class to perform time operations.
+ */
+@Component
+class TimeService {
+    /**
+     * Convert String to LocalTime by pattern HH:mm.
+     *
+     * @throws IllegalArgumentException if String doesn't satisfy the pattern
+     */
+    fun convert(source: String?): LocalTime {
+        return try {
+            LocalTime.parse(source, formatter)
+        } catch (e: DateTimeParseException) {
+            logger.warn("Failed to parse to LocalDate", e)
+            throw IllegalArgumentException("Illegal data")
+        }
```

Comment:
use kotlin version of try

Reply:
You mean use the .use{ LocalTime.parse(source, formatter) } method? If yes, then I don't understand how I can replace the DateTimeParseException error with IllegalArgumentException, and I need it because my logic is linked to it. I can't find any other versions of the try block in kotlin.